File: src/main/java/core/constants/player/PlayerSkill.java
package core.constants.player;

import core.model.HOVerwaltung;


public enum  PlayerSkill {

//	private static final String[] languageKeys = {
//			"ls.player.skill.keeper",
//			"ls.player.skill.defending",
//			"ls.player.skill.winger",
//			"ls.player.skill.playmaking",
//			"ls.player.skill.scoring",
//			"ls.player.skill.passing",
//			"ls.player.skill.stamina",
//			"ls.player.form",
//			"ls.player.skill.setpieces",
//			"ls.player.experience",
//			"ls.player.leadership",
//			"ls.player.loyalty"
//	};

	KEEPER(0),
	DEFENDING(1),
	WINGER(2),
	PLAYMAKING(3),
	SCORING(4),
	PASSING(5),
	STAMINA(6),
	FORM(7),
	SETPIECES(8),
	EXPERIENCE(9),
	LEADERSHIP(10),
	LOYALTY(11);

	private final int id;

	PlayerSkill(int id) {
		this.id = id;
	}

	public int toInt() {
		return id;
	}

	public String getLanguageString() {
		var b = new StringBuilder("ls.player.");
		switch (this) {
			case KEEPER, DEFENDING, WINGER, PLAYMAKING, SCORING, PASSING, STAMINA, SETPIECES -> b.append("skill.");
		}
		b.append(super.toString().toLowerCase());
		return HOVerwaltung.instance().getLanguageString(b.toString());
	}

	public static PlayerSkill fromInteger(Integer i) {
		if (i != null) {
			for (var s : PlayerSkill.values()) {
				if (s.id == i) {
					return s;
				}
			}
		}
		return null;
	}

	public String getXMLElementName(){
		return switch (this){
			case KEEPER -> "Keeper";
			case DEFENDING -> "Defender";
			case PLAYMAKING -> "Playmaker";
			case WINGER -> "Winger";
			case PASSING -> "Passing";
			case SCORING -> "Scorer";
			case SETPIECES -> "SetPieces";
			default -> "unknown";
		};
	}
}

File: src/main/java/core/constants/player/PlayerSkill.kt
package core.constants.player

import core.model.HOVerwaltung

object PlayerSkill {
    private val languageKeys = arrayOf(
        "ls.player.skill.keeper",
        "ls.player.skill.defending",
        "ls.player.skill.winger",
        "ls.player.skill.playmaking",
        "ls.player.skill.scoring",
        "ls.player.skill.passing",
        "ls.player.skill.stamina",
        "ls.player.form",
        "ls.player.skill.setpieces",
        "ls.player.experience",
        "ls.player.leadership",
        "ls.player.loyalty"
    )
    const val KEEPER = 0
    const val DEFENDING = 1
    const val WINGER = 2
    const val PLAYMAKING = 3
    const val SCORING = 4
    const val PASSING = 5
    const val STAMINA = 6
    const val FORM = 7
    const val SET_PIECES = 8
    const val EXPERIENCE = 9
    const val LEADERSHIP = 10
    const val LOYALTY = 11
    @JvmStatic
    fun toString(skill: Int): String {
        return if (skill in KEEPER..LOYALTY)
            HOVerwaltung.instance().getLanguageString(languageKeys[skill])
        else HOVerwaltung.instance().getLanguageString("Unbestimmt")
    }
}
